csrc/kernel_cache.h

```diff
@@ -124,7 +124,7 @@ class FusionKernelRuntime {
     std::lock_guard<std::mutex> guard(mutex_);
     return std::all_of(
         executors_.begin(), executors_.end(), [](const auto& executor) {
-          return executor.isCompiled();
+          return executor.isCompiled() || executor.isCompilationSkipped();
```

Comment:
nitpick: This feels a little bit confusing to me.

i.e. `FusionKernelRuntime::isCompiled` means something different from `FusionExecutor::isCompiled`. (similarly we have `FusionExecutorCache::isCompiled`.

We don't have to do it in this PR, but maybe another renaming PR to clean up the `isCompiled` vs `isCompilationSkipped` in `FusionExecutor`.

Reply:
I should rename it to `FusionKernelRuntime::isCompiledOrCompilationSkipped`.
This function is used to check if compilation was already attempted which is true if either the executors are all compiled or compilation was skipped for them.

Reply:
Any other naming suggestions for this function?

Reply:
1. isCompilationSkipped() --- Returns True when we use the Aten function and `FusionExecutor` is never compiled.
2. isCompiled() --- Returns False when `FusionExecutor` has yet to be compiled.

Maybe we should rename `isCompilationSkipped()` to `isAtenExecuted()` or `isEagerModeExecutor()`

Reply:
I agree with Ryan's suggestion. We can keep `isCompiled` to denote if compilation was attempted -- either a kernel was compiled or we are using Aten evaluation. Additionally, `isCompilationSkipped` is renamed to `isEagerModeExecutor`.

In this naming scheme, `isCompiled` always means whether or not we attempted to compile a fusion.

Wdyt @jjsjann123?

Reply:
When `isEagerModeExecutor` is `false`, we're using the nvfuser executor.

`isCompiled` implies that we have a cubin for the given fusion. There is an implicit assumption that the nvfuser executor is being used. 

I'd consider renaming `isCompiled` to `isCubinCompiled`, but leaving it alone is also fine.

Reply:
Okay, so the renaming is 
1. `FusionExecutor::isCompiled` -> `FusionExecutor::isCubinCompiled`
2. `FusionExecutor::isCompilationSkipped` -> `FusionExecutor::isEagerModeExecutor`
3. `FusionKernelRuntime/FusionExecutorCache::isCompiled` -> no change. Returns true if either of the above are true indicating that compilation was attempted.

I'll go with this naming if it clears the confusion. @rdspring1 @jjsjann123

Reply:
sounds very reasonable to me.